webapp/channels/src/components/suggestion/emoticon_provider.test.tsx

```diff
@@ -48,14 +53,18 @@ describe('components/EmoticonProvider', () => {
     it('should order suggested emojis', () => {
         const pretext = ':thu';
         const recentEmojis = ['smile'];
-        getEmojiMap.mockReturnValue(emojiMap);
-        getRecentEmojisNames.mockReturnValue(recentEmojis);
+        (getEmojiMap as unknown as jest.Mock).mockReturnValue(emojiMap);
+        (getRecentEmojisNames as unknown as jest.Mock).mockReturnValue(
+            recentEmojis,
+        );
 
         emoticonProvider.handlePretextChanged(pretext, resultsCallback);
         expect(resultsCallback).toHaveBeenCalled();
         const args = resultsCallback.mock.calls[0][0];
-        const results = args.items.filter((item) => item.name.indexOf('skin') === -1);
-        expect(results.map((item) => item.name)).toEqual([
+        const results = args.items.filter(
+            (item: Emoji) => item.name.indexOf('skin') === -1,
```

Comment:
Why the type definition is needed? In theory, the type of `items` should be defined, and the type in the function arguments should be properly inferred. Or am I missing something?

Reply:
By default, the values of args.items is any. The value of args is set via jest function, as on line 64. Hence, typescript was complaining about this until I explicitly defined the type of the item variable.